src/components/app.js

```diff
@@ -8,9 +8,8 @@ import TodoItem from './todo-item';
 @connect(reduce, bindActions(actions))
 export default class App extends Component {
 	addTodos = () => {
-		let { text } = this.state;
-		this.setState({ text: '' });
-		this.props.addTodo(text);
+		this.props.addTodo(this.input.value);
+		this.input.value = '';
```

Comment:
Good point! `setState()` worked fine with a manual handler. Adding a todo item now uses the state value again.